### PR TITLE
Fix for #5816

### PR DESCRIPTION
**Pull request by vinayak25**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [NA] Tests for the changes have been added (for bug fixes / features)
- [NA] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?
method `applyLogger` inside nest-factory.ts method is checking for `logger` attribute existence and not it's passed value.
Issue Number: #5816 

## What is the new behavior?
method `applyLogger` inside nest-factory.ts method will now check for `logger` attribute value.

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

### CODE REVIEW

File: packages/core/nest-factory.ts [hunk 1]
```diff
@@ -218,7 +218,7 @@ export class NestFactoryStatic {
     if (!options) {
       return;
     }
-    !isNil(options.logger) && Logger.overrideLogger(options.logger);
+    !options.logger && Logger.overrideLogger(options.logger);
```
Comment (kamilmysliwiec) [inline packages/core/nest-factory.ts:221]: This will break the default scenario in which `options.logger` is `undefined`.

Comment (vinayak25) [reply packages/core/nest-factory.ts:221]: incase options.logger is undefined, it will automatically evaluate to `true`

Comment (kamilmysliwiec) [reply packages/core/nest-factory.ts:221]: And then, `undefined` will be passed down to the `Logger.overrideLogger` method which will disable the logger.

Comment (vinayak25) [reply packages/core/nest-factory.ts:221]: Got it. Changing it. Thanks for the suggestion

Comment (vinayak25) [reply packages/core/nest-factory.ts:222]: Fixed in the new commit. Please check now

**Comment by kamilmysliwiec** [comment]
Thanks!

**Pull request merged**